tests/stub/driver_parameters/test_bookmark_manager.py

```diff
@@ -0,0 +1,598 @@
+        self.assert_run(run_requests[0])
+        self.assert_run(
+            run_requests[1],
+            bookmarks=["bm1"]
+        )
+        self.assert_run(
+            run_requests[2],
+            bookmarks=["bm1", "adb:bm4"]
+        )
+        self.assert_run(
+            run_requests[3],
+            bookmarks=["bm2", "adb:bm4"]
+        )
+
+    def test_should_resolve_database_name_with_system_bookmarks(self):
+        self._start_server(self._router, "router_with_db_name.script")
+        self._start_server(self._server, "transaction_chaining.script")
+
+        uri = "neo4j://%s" % self._router.address
+        auth = types.AuthorizationToken("basic", principal="neo4j",
+                                        credentials="pass")
+        self._driver = Driver(
+            self._backend,
+            uri, auth,
+            bookmark_manager_config=Neo4jBookmarkManagerConfig(
+                initial_bookmarks={"system": ["sys:bm1"]}
+            )
+        )
+
+        s1 = self._driver.session("w")
+        tx1 = s1.begin_transaction({"order": "1st"})
+        tx1.run("RETURN 1 as n").consume()
+        tx1.commit()
+        s1.close()
+
+        s2 = self._driver.session("w")
+        tx2 = s2.begin_transaction({"order": "2nd"})
+        tx2.run("RETURN 1 as n").consume()
+        tx2.commit()
+        s2.close()
+
+        route_requests = self._router.get_requests("ROUTE")
+        begin_requests = self._server.get_requests("BEGIN")
+
+        self.assertGreaterEqual(len(route_requests), 1)
+        self.assertEqual(len(begin_requests), 2)
+
+        self.assert_route(route_requests[0], bookmarks=["sys:bm1"])
+
+        self.assert_begin(
+            begin_requests[0],
+            bookmarks=["sys:bm1"]
+        )
+        self.assert_begin(
+            begin_requests[1],
+            bookmarks=["sys:bm1", "bm1"]
+        )
+
+    def _start_server(self, server, script):
+        server.start(self.script_path(script),
+                     vars_={"#HOST#": self._router.host})
+
+    def assert_begin(self, line: str, bookmarks=None):
+        if bookmarks is None:
+            bookmarks = []
+        begin_prefix = "BEGIN "
+        self.assertTrue(
+            line.startswith(begin_prefix),
+            "Line should start with begin"
+        )
+        begin_properties = json.loads(line[len(begin_prefix):])["{}"]
+        if not bookmarks:
+            self.assertFalse(
+                "bookmarks" in begin_properties,
+                "bookmarks should not be in the begin"
+            )
+        else:
+            self.assertEqual(
+                sorted(bookmarks),
+                sorted(begin_properties.get("bookmarks", []))
+            )
+
+    def assert_route(self, line: str, bookmarks=None):
+        if bookmarks is None:
+            bookmarks = []
+        route_prefix = "ROUTE "
+        self.assertTrue(
+            line.startswith(route_prefix),
+            "Line should start with ROUTE"
+        )
```

Comment:
```suggestion
        self.assertTrue(line.startswith("ROUTE "))
```
You don't _have to_ change this, but I find this line of code is as good as stating `"Line should start with ROUTE"` explicitly + it's a fair bit shorter (same for `run_prefix = "RUN "` and `begin_prefix = "BEGIN "`).

Reply:
I prefer have a more human readable log.